js/register_user_website/validate_input.js

```diff
@@ -0,0 +1,134 @@
+function validate_fields() {
+    hide_alert_all();
+    var email = document.getElementById("email_input").value;
+    var username = document.getElementById("username_input").value;
+    var password = document.getElementById("password_input").value;
+    if (password == "" || username == "" || email == "") {
+        fill_warnings("All fields need to be filled out by you.");
+    }
+    else if (!check_username(username)) {
+        return;
+    }
+    else if (!check_email_length(email)) {
+        return;
+    }
+    else if (!check_password(password)) {
+        return;
+    }
+    else {
+        check_duplicates_email(email);
+        check_duplicates_username(username);
+    }
+}
+function hide_alert_with_button(event) {
+    event.target.parentNode.parentNode.setAttribute("style", "visibility: hidden !important;");
+    event.target.previousElementSibling.innerHTML = "";
+}
+function hide_alert_all() {
+    var alerts_text = document.getElementsByClassName("d-flex align-items-center justify-content-center m-2");
+    for (var i = 0; i < alerts_text.length; i++) {
+        alerts_text[i].setAttribute("style", "visibility: hidden !important;");
+        alerts_text[i].firstElementChild.firstElementChild.nextElementSibling.textContent = "";
+    }
+}
+function fill_warnings(warning_text) {
+    var alerts_text = document.getElementsByClassName("alert_text");
+    //Check if alert is already being displayed
+    for (var i = 0; i < alerts_text.length; i++) {
+        if (alerts_text[i].textContent == warning_text) {
+            return;
+        }
+    }
+    for (var i = 0; i < alerts_text.length; i++) {
+        if (alerts_text[i].textContent == "") {
+            alerts_text[i].textContent = warning_text;
+            alerts_text[i].parentElement.parentElement.setAttribute("style", "visibility: visible !important;");
+            break;
```

Comment:
gehört der teil nicht in den else teil von der oberen for schleife? Dann braucht man nicht 2 mal die for schleife

Reply:
Habe es umgeschrieben, gute Idee 🥇